drivers/display/display_st7735r.c

```diff
@@ -530,11 +491,10 @@ static const struct display_driver_api st7735r_api = {
 
 #define ST7735R_INIT(inst)							\
 	const static struct st7735r_config st7735r_config_ ## inst = {		\
-		.bus = SPI_DT_SPEC_INST_GET(					\
-			inst, SPI_OP_MODE_MASTER | SPI_WORD_SET(8) |		\
-			SPI_HOLD_ON_CS | SPI_LOCK_ON, 0),			\
-		.cmd_data = GPIO_DT_SPEC_INST_GET(inst, cmd_data_gpios),	\
-		.reset = GPIO_DT_SPEC_INST_GET_OR(inst, reset_gpios, {}),	\
+		.mipi_dev = DEVICE_DT_GET(DT_INST_PARENT(inst)),		\
+		.dbi_config = MIPI_DBI_CONFIG_DT_INST(inst,			\
+				SPI_OP_MODE_MASTER | SPI_WORD_SET(8) |		\
```

Comment:
SPI_WORD_SET(8)
It will be better if you set correct SPI data size depend on the selected mode (8 for 4wire, 9 for 3wire).

Reply:
I think this makes sense. Doing so will also allow dropping the `tmp_cfg` within the MIPI DBI driver.